plugin/trino-ignite/src/main/java/io/trino/plugin/ignite/IgniteJdbcClient.java

```diff
@@ -0,0 +1,655 @@
+                switch (property) {
+                    case AFFINITY_KEY_PROPERTY:
+                        List<String> affinityKeys = extractColumnNamesFromOrderingScheme(resultSet.getString("AFK"));
+                        if (!affinityKeys.isEmpty()) {
+                            String affinityKey = affinityKeys.get(0);
+                            checkArgument(ImmutableSet.copyOf(primaryKeys).contains(affinityKey), "Table affinity key should be one of the primary key");
+                            properties.put(property, affinityKey);
+                        }
+                        break;
+                    case BACKUPS_PROPERTY:
+                        int backups = resultSet.getInt(property.toUpperCase(ENGLISH));
+                        if (backups > 0) {
+                            properties.put(property, backups);
+                        }
+                        break;
+                    case TEMPLATE_PROPERTY:
+                        Optional.ofNullable(resultSet.getString(property.toUpperCase(ENGLISH)))
+                                .map(IgniteTableProperties.IgniteTemplateType::valueOf)
+                                .ifPresent(value -> properties.put(property, value));
+                        break;
+                    case WRITE_SYNCHRONIZATION_MODE_PROPERTY:
+                        Optional.ofNullable(resultSet.getString(property.toUpperCase(ENGLISH)))
+                                .map(IgniteTableProperties.IgniteWriteSyncMode::valueOf)
+                                .ifPresent(value -> properties.put(property, value));
+                        break;
+                    case CACHE_NAME_PROPERTY:
+                    case CACHE_GROUP_PROPERTY:
+                    case DATA_REGION_PROPERTY:
+                        Optional.ofNullable(resultSet.getString(property.toUpperCase(ENGLISH)))
+                                .ifPresent(value -> properties.put(property, value));
+                        break;
+                    default:
+                        throw new IllegalStateException("Unexpected value: " + property);
+                }
+            }
+        }
+        catch (SQLException e) {
+            throw new TrinoException(JDBC_ERROR, e);
+        }
+        return properties.build();
+    }
+
+    // extract result from format: "ID" ASC, "CITY_ID" ASC
+    private List<String> extractColumnNamesFromOrderingScheme(String row)
+    {
+        ImmutableList.Builder<String> builder = ImmutableList.builder();
+        if (isNullOrEmpty(row)) {
+            return builder.build();
+        }
+        for (String key : SPLITTER.split(row)) {
+            int left = key.indexOf("\"") + 1;
+            int right = key.lastIndexOf("\"");
+            builder.add(key.substring(left, right).toLowerCase(ENGLISH));
+        }
+        return builder.build();
+    }
+
+    private static SliceWriteFunction uuidWriteFunction()
```

Comment:
Helper functions were added for handling UUIDs recently. You can take a look at changes in https://github.com/trinodb/trino/pull/8942 and apply them here.

Reply:
Oh, I should remove this code, we can not support uuid now, coz the type in Ignite is "OTHER".